### PR TITLE
OCPBUGS-8512: fix admission webhook CA injection

### PR DESCRIPTION
**Pull request by stlaz**
This PR fixes the problem outlined in https://github.com/openshift/service-ca-operator/pull/207. It unifies the webhook injectors logic and uses the original PR bits for the e2e tests.

### CODE REVIEW

File: pkg/controller/cabundleinjector/admissionwebhook.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+	return err
+}
+
+func newMutatingWebhookInjectorConfig(config *caBundleInjectorConfig) controllerConfig {
+	informer := config.kubeInformers.Admissionregistration().V1().MutatingWebhookConfigurations()
+	syncer := &webhookCABundleInjector[admissionregv1.MutatingWebhookConfiguration]{
+		webhookConfigType:         "mutatingwebhookconfiguration",
+		client:                    config.kubeClient.AdmissionregistrationV1().MutatingWebhookConfigurations(),
+		lister:                    informer.Lister(),
+		webhookConfigAccessorInit: newMutatingWebhookAccessor,
```
Comment (s-urbaniak) [inline pkg/controller/cabundleinjector/admissionwebhook.go:133]: let's call this `newWebhookAccessor`, this naming provides a nice abstraction to the concrete implementations `newMutatingWebhookAccessor` and `newValidatingWebhookAccessor`.

Comment (s-urbaniak) [reply pkg/controller/cabundleinjector/admissionwebhook.go:133]: note: initialization is different from construction and usually comes after it.

**Comment by sam6258** [comment]
Thanks, @stlaz. I would close out #207 myself, but don't have permissions configured in prow/repo

https://github.com/openshift/service-ca-operator/pull/207#issuecomment-1726230431

**Pull request merged**